src/prune.ts

```diff
@@ -0,0 +1,247 @@
+import { join } from 'path'
+import Promise = require('any-promise')
+import { EventEmitter } from 'events'
+import { Emitter } from './interfaces'
+import { resolveTypeDependencies, Options as DependencyOptions } from './lib/dependencies'
+import { findProject } from './utils/find'
+import { readFile, readConfig, writeFile, isDirectory, readSubDirs, rimraf } from './utils/fs'
+import {
+  CONFIG_FILE,
+  TYPINGS_DIR,
+  DTS_BROWSER_FILE,
+  DTS_MAIN_FILE,
+  MAIN_TYPINGS_DIR,
+  BROWSER_TYPINGS_DIR,
+  DEFINITIONS_DIR,
+  AMBIENT_DEFINITIONS_DIR
+} from './utils/config'
+
+const REFERENCE_REGEXP = /^\/\/\/[ \t]*<reference[ \t]+path[ \t]*=("|')(.*?)\1.*?\/>[ \t]*\r?\n?/
```

Comment:
What are you trying to do exactly? I wouldn't be using the RegExp directly myself, I'd be using the functions exposed. https://github.com/typings/core/blob/master/src/utils/references.ts#L36